source/extensions/filters/http/header_to_metadata/header_to_metadata_filter.cc

```diff
@@ -182,20 +207,17 @@ void HeaderToMetadataFilter::writeHeaderToMetadata(Http::HeaderMap& headers,
         ENVOY_LOG(debug, "value is empty, not adding metadata");
       }
 
-      if (rule.remove()) {
+      if (proto_rule.remove()) {
         headers.remove(header);
       }
-    } else if (rule.has_on_header_missing()) {
+    } else if (proto_rule.has_on_header_missing()) {
       // Add metadata for the header missing case.
-      const auto& keyval = rule.on_header_missing();
+      const auto& keyval = proto_rule.on_header_missing();
 
-      if (!keyval.value().empty()) {
-        const auto& nspace = decideNamespace(keyval.metadata_namespace());
-        addMetadata(structs_by_namespace, nspace, keyval.key(), keyval.value(), keyval.type(),
-                    keyval.encode());
-      } else {
-        ENVOY_LOG(debug, "value is empty, not adding metadata");
-      }
+      ASSERT(!keyval.value().empty());
```

Comment:
```
    // When used for a :ref:`on_header_missing <envoy_v3_api_field_extensions.filters.http.header_to_metadata.v3.Config.Rule.on_header_missing>`
    // case, a non-empty value must be provided otherwise no metadata is added.
    string value = 3 [(udpa.annotations.field_migrate).oneof_promotion = "value_type"];
```

Reply:
Granted, at config creation time we should reject `on_header_missing` rules with empty value. I'll add that to avoid crashing debug code.

Reply:
@zuercher https://github.com/envoyproxy/envoy/pull/11698/commits/f6e422dcbf9d08f39c453a0ad7c73b91e4137a7c should fix this?

Reply:
That works.